core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -915,11 +929,39 @@ open class TileInfo {
             else improvement.getTurnsToBuild(civInfo, unit)
     }
 
+    /** Clears [improvementInProgress] and [turnsToImprovement] */
     fun stopWorkingOnImprovement() {
         improvementInProgress = null
         turnsToImprovement = 0
     }
 
+    /** Sets tile improvement to pillaged (without prior checks for validity)
+     *  and ensures that matching [UniqueType.CreatesOneImprovement] queued buildings are removed. */
+    fun setPillaged() {
+        // http://well-of-souls.com/civ/civ5_improvements.html says that naval improvements are destroyed upon pillage
+        //    and I can't find any other sources so I'll go with that
+        if (isLand) {
+            // Setting turnsToImprovement might interfere with UniqueType.CreatesOneImprovement
+            removeCreatesOneImprovementMarker()
```

Comment:
What happens in this case if the city completes construction?

Reply:
Can't since the marker removal also removes the queued building. I didn't expand on that in the comment as the Kdoc says it and it's only twelve lines down to see how.